core/src/main/java/bisq/core/trade/protocol/bsq_swap/tasks/seller/ProcessTxInputsMessage.java

```diff
@@ -45,41 +47,53 @@ public ProcessTxInputsMessage(TaskRunner<BsqSwapTrade> taskHandler, BsqSwapTrade
     @Override
     protected void run() {
         try {
-            runInterceptHook();
+            TxInputsMessage message = checkNotNull((TxInputsMessage) protocolModel.getTradeMessage());
+            checkNotNull(message);
 
-            TxInputsMessage data = checkNotNull((TxInputsMessage) protocolModel.getTradeMessage());
-            checkNotNull(data);
+            List<RawTransactionInput> inputs = message.getBsqInputs();
+            checkArgument(!inputs.isEmpty(), "Buyers BSQ inputs must not be empty");
 
-            List<RawTransactionInput> inputs = data.getBsqInputs();
-            checkArgument(!inputs.isEmpty(), "inputs must not be empty");
             long sumInputs = inputs.stream().mapToLong(rawTransactionInput -> rawTransactionInput.value).sum();
-            // If taker is buyer the inputs are BSQ, otherwise BTC
-            checkArgument(sumInputs >= BsqSwapCalculation.getBuyersRequiredBsqInputs(trade).getValue(),
-                    "Buyers BSQ inputs are not sufficient");
-
-            long change = data.getBsqChange();
+            long buyersBsqInputAmount = BsqSwapCalculation.getBuyersBsqInputValue(trade, getBuyersTradeFee()).getValue();
+            checkArgument(sumInputs >= buyersBsqInputAmount,
+                    "Buyers BSQ input amount do not match our calculated required BSQ input amount");
+
+            DaoFacade daoFacade = protocolModel.getDaoFacade();
+            long numValidBsqInputs = inputs.stream()
+                    .map(input -> new TxOutputKey(txIdOfRawInputParentTx(input), (int) input.index))
+                    .filter(daoFacade::isTxOutputSpendable)
+                    .count();
+            checkArgument(inputs.size() == numValidBsqInputs,
+                    "Some of the buyers BSQ inputs are not from spendable BSQ utxos according to our DAO state data.");
+
+            long change = message.getBsqChange();
```

Comment:
Why trust the peer change amount rather than calculating it from BSQ inputs and BSQ payout? I don't see this value being verified later on.